be/src/exprs/vectorized/array_functions.tpp

```diff
@@ -105,6 +105,303 @@ private:
+                                          length_column->get_data()[i]);
+                    }
+                }
+            } else {
+                ColumnPtr offset_column = ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[1]);
+                ColumnPtr length_column = ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[2]);
+
+                const auto* src_nullable_column = down_cast<const NullableColumn*>(src_column.get());
+                const auto* src_data_column = down_cast<const ArrayColumn*>(src_nullable_column->data_column().get());
+                auto& dest_nullable_column = down_cast<NullableColumn&>(*dest_column);
+                auto& dest_null_data = down_cast<NullableColumn&>(*dest_column).null_column_data();
+                auto& dest_data_column = down_cast<ArrayColumn&>(*dest_nullable_column.data_column());
+
+                dest_null_data = src_nullable_column->immutable_null_column_data();
+                dest_nullable_column.set_has_null(src_nullable_column->has_null());
+
+                if (src_nullable_column->has_null() || offset_column->has_null() || length_column->has_null()) {
+                    for (size_t i = 0; i < chunk_size; i++) {
+                        if (!src_nullable_column->is_null(i) && !offset_column->is_null(i) &&
+                            !length_column->is_null(i)) {
+                            _array_slice_item(*src_data_column, i, &dest_data_column, offset_column->get(i).get_int64(),
+                                              length_column->get(i).get_int64());
+                        } else {
+                            dest_data_column.append_default();
+                        }
+                    }
+                } else {
+                    for (size_t i = 0; i < chunk_size; i++) {
+                        _array_slice_item(*src_data_column, i, &dest_data_column, offset_column->get(i).get_int64(),
+                                          length_column->get(i).get_int64());
+                    }
+                }
+            }
+        } else {
+            const auto* src_data_column = down_cast<const ArrayColumn*>(src_column.get());
+            auto* dest_data_column = down_cast<ArrayColumn*>(dest_column.get());
+
+            ColumnPtr offset_column = ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[1]);
+            ColumnPtr length_column = ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[2]);
+
+            if (offset_column->has_null() || length_column->has_null()) {
+                for (size_t i = 0; i < chunk_size; i++) {
+                    if (!offset_column->is_null(i) && !length_column->is_null(i)) {
+                        _array_slice_item(*src_data_column, i, dest_data_column, offset_column->get(i).get_int64(),
+                                          length_column->get(i).get_int64());
+                    } else {
```

Comment:
if src column is not nullable, the offset is null, the result is ?

Reply:
If anyone is null, result is null